tests/test_dimension.py

```diff
@@ -542,7 +542,17 @@ def test_basic(self):
 
         eqns = [Eq(u.forward, u + 1.), Eq(u2.forward, u2 + 1.), Eq(usave, u)]
         op = Operator(eqns)
-        op.apply(t_M=nt-2)
+        op.apply()
+        assert np.all(np.allclose(u.data[(nt-1) % 3], nt-1))
+        assert np.all([np.allclose(u2.data[i], i) for i in range(nt)])
+        assert np.all([np.allclose(usave.data[i], i*factor)
+                      for i in range((nt+factor-1)//factor)])
+
+        # Check overrides
```

Comment:
is this worth a separate test *but* with just equation (so simpler than this one in terms of symbols and equations) *and* the override? might help future debugging

Reply:
done